pkg/controller/certificates/sync.go

```diff
@@ -409,12 +412,23 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 	reason := apiutil.CertificateRequestReadyReason(existingReq)
 	// Determine the status reason of the CertificateRequest and process accordingly
 	switch reason {
-	// If the CertificateRequest exists but has failed then we check the failure
-	// time. If the failure time doesn't exist or is over an hour in the past
-	// then delete the request so it can be re-created on the next sync. If the
-	// failure time is less than an hour in the past then schedule this owning
-	// Certificate for a re-sync in an hour.
+	// If the CertificateRequest exists but has failed then we check the
+	// InvalidRequest condition as well as failure time.
+	// - If the CertificateRequest condition is present and has the status of
+	//   "True" then do not attempt to retry the CertificateRequest. Else we can
+	//   retry.
+	// - If the failure time doesn't exist or is over an hour in the past then
+	//   delete the request so it can be re-created on the next sync. If the
+	//   failure time is less than an hour in the past then schedule this owning
+	//   Certificate for a re-sync in an hour.
 	case cmapi.CertificateRequestReasonFailed:
+		if apiutil.CertificateRequestHasInvalidRequest(existingReq) {
```

Comment:
I think invalid request is a distinct 'case' altogether, and should be checked before the 'Ready' condition is checked altogether - if the request is *invalid* then we should not proceed any further, as changes are needed to make the request valid.

Having one condition only effective if another condition is set to a specific value is confusing and could well be error prone in future.